src/renderer/src/components/Popups/SelectModelPopup/popup.tsx

```diff
@@ -185,92 +153,52 @@ const PopupContainer: React.FC<Props> = ({ model, resolve, modelFilter }) => {
         isSelected: false
       })
 
-      items.push(...filteredModels.map((m) => createModelItem(m, p, pinnedModels.includes(getModelUniqId(m)))))
+      items.push(...filteredModels.map((m) => createModelItem(m, p, pinnedModelIds.has(getModelUniqId(m)))))
     })
 
-    // 移除第一个分组标题，使用 sticky group banner 替代，模拟 sticky 效果
-    if (items.length > 0 && items[0].type === 'group') {
-      firstGroupRef.current = items[0]
-      items.shift()
-    } else {
-      firstGroupRef.current = null
-    }
-    return items
+    // 获取可选择的模型项（过滤掉分组标题）
+    const modelItems = items.filter((item) => item.type === 'model') as FlatListItem[]
+    return { listItems: items, modelItems }
   }, [searchText.length, pinnedModels, providers, modelFilter, createModelItem, t, getFilteredModels])
 
-  // 获取可选择的模型项（过滤掉分组标题）
-  const modelItems = useMemo(() => {
-    return listItems.filter((item) => item.type === 'model')
-  }, [listItems])
-
-  // 当搜索文本变化时更新滚动触发器
-  useEffect(() => {
-    searchChanged(searchText)
-  }, [searchText, searchChanged])
-
-  // 基于滚动位置更新sticky分组标题
-  const updateStickyGroup = useCallback(
-    (scrollOffset?: number) => {
-      if (listItems.length === 0) {
-        stickyGroup && setStickyGroup(null)
-        return
-      }
+  const listHeight = useMemo(() => {
+    return Math.min(PAGE_SIZE, listItems.length) * ITEM_HEIGHT
+  }, [listItems.length])
 
-      let newStickyGroup: FlatListItem | null = null
+  // 处理程序化滚动（加载、搜索开始、搜索清空）
+  useLayoutEffect(() => {
+    if (loading) return
 
-      // 基于滚动位置计算当前可见的第一个项的索引
-      const estimatedIndex = Math.floor((scrollOffset ?? lastScrollOffset) / ITEM_HEIGHT)
+    if (preventScrollToIndex.current) {
+      preventScrollToIndex.current = false
+      return
+    }
 
-      // 从该索引向前查找最近的分组标题
-      for (let i = estimatedIndex - 1; i >= 0; i--) {
-        if (i < listItems.length && listItems[i]?.type === 'group') {
-          newStickyGroup = listItems[i]
-          break
-        }
-      }
+    let targetItemKey: string | undefined
 
-      // 找不到则使用第一个分组标题
-      if (!newStickyGroup) newStickyGroup = firstGroupRef.current
+    // 启动搜索时，滚动到第一个 item
+    if (searchText) {
+      targetItemKey = modelItems[0]?.key
+    }
+    // 初始加载或清空搜索时，滚动到 selected item
+    else {
+      targetItemKey = modelItems.find((item) => item.isSelected)?.key
+    }
 
-      if (stickyGroup?.key !== newStickyGroup?.key) {
-        setStickyGroup(newStickyGroup)
+    if (targetItemKey) {
+      setFocusedItemKey(targetItemKey)
+      const index = listItems.findIndex((item) => item.key === targetItemKey)
+      if (index >= 0) {
+        // FIXME: 手动计算偏移量给 scroller 增加了 padding 之后使用 scrollToIndex 不能
+        // 准确滚动到 item 中心，但是我们又需要 padding 来改善体验。
+        const targetScrollTop = index * ITEM_HEIGHT - listHeight / 2
```

Comment:
我看scroller在y方向上没有padding？这里是滚动到稍微偏下的位置，因为对齐的位置是item的上边界。

```suggestion
        const targetScrollTop = index * ITEM_HEIGHT - listHeight / 2 + ITEM_HEIGHT / 2
```

Reply:
不是样式上的 padding，是虚拟列表滚动 padding，我改一下注释